### ISSUE TITLE
xCAT won't install on RHEL8 system with FIPS enabled.

### ISSUE THREAD

**Issue by bviviano** [xcat2/xcat-core#7158]
I am attempting to install xCAT 2.16.3 onto a new RHEL 8.5 server with FIPS installed, but it is failing with:

Error: Transaction test error:
  package xCAT-genesis-base-x86_64-2:2.14.5-snap201811190037.noarch does not verify: no digest
  package xCAT-genesis-scripts-x86_64-1:2.16.3-snap202111100958.noarch does not verify: no digest
  package xCAT-genesis-scripts-ppc64-1:2.16.3-snap202111100958.noarch does not verify: no digest
  package syslinux-xcat-3.86-2.noarch does not verify: no digest
  package perl-Net-HTTPS-NB-0.14-2.noarch does not verify: no digest
  package grub2-xcat-2.02-0.76.el7.1.snap201905160255.noarch does not verify: no digest
  package elilo-xcat-3.14-6.noarch does not verify: no digest
  package xCAT-buildkit-4:2.16.3-snap202111100958.noarch does not verify: no digest
  package perl-xCAT-4:2.16.3-snap202111100958.noarch does not verify: no digest
  package xCAT-client-4:2.16.3-snap202111100958.noarch does not verify: no digest
  package xCAT-probe-4:2.16.3-snap202111100958.noarch does not verify: no digest
  package xCAT-server-4:2.16.3-snap202111100958.noarch does not verify: no digest
  package xCAT-2.16.3-snap202111100958.x86_64 does not verify: no digest

This is traced back to this KB article at Red Hat's site:

https://access.redhat.com/solutions/4460971

Resolution
- Have package rebuilt with rpm >= 4.14
- Or disable FIPS

Operationally, we can't disable FIPS so the only possible work around right now is to manually download each RPM and install using

`rpm -ivh --nodigest --nofiledigest`

which is problematic.  Please consider updating the RPM's to support the use of rpmbuild included with RHEL8/CentOS8 that resolves this issue.

**Comment by bviviano** [comment]
Wanted to add a followup, in case others have this issue.

https://access.redhat.com/discussions/6333871

The steps in the above discussion work, as a work around, but I do request that versions of the RPM's for RHEL8/CentOS8 be provided built with 4.14+.

**Comment by besawn** [comment]
xCAT does not officially support installation with FIPS enabled. Have you been able to successfully complete an installation with FIPS enabled on prior versions of xCAT or with prior versions of RHEL?

**Comment by bviviano** [comment]
Yes, we've previously installed xCAT 2.x on several RHEL6 and 7 with FIPS enabled clusters.  We currently have a cluster setup based on RHEL 7.9 with xCAT 2.15.1 that was deployed from RHEL ISO with fips=1 in the kickstart.

The issue isn't FIPS specifically, but that RHEL8 w/FIPS requires RPM's to be built with rpmbuild 4.14 or later.

I can confirm this from my own testing, if I create a simple spec and build it to RPM on RHEL7 (rpmbuild 4.11) and try and deploy it on a RHEL8 system with FIPS, I get the 

`package does not verify: no digest`

error, if I take the same spec file and build it on RHEL 8.5 (rpmbuild 4.14.3) I have no issues.

**Comment by viniciusferrao** [comment]
Build system have changed for 2.18 so this may be partially addressed. I'll take a look on this.

There is still demand for that @bviviano?

We are willing to fix that.

**Comment by bviviano** [comment]
Yes, we'd still like to have RPMs built with SHA256 vs. SHA1.

**Comment by viniciusferrao** [comment]
@dhilst can you look at some packages in `xcat-dep`? I think only it missed the some packages from it.

**Comment by viniciusferrao** [comment]
@bviviano this is fixed already in newer xCAT version. You should be able to run the system in FIPS mode now.

| | AlmaLinux 8.10 | AlmaLinux 9.8 | AlmaLinux 10.2 |
|---|---|---|---|
| `crypto.fips_enabled` | 1 | 1 | 1 |
| crypto policy | FIPS | FIPS | FIPS |
| kernel cmdline | `fips=1 boot=UUID=...` | `fips=1 boot=UUID=...` | `fips=1 boot=UUID=...` |
| rpm version | 4.14.3 | 4.16.1.3 | 4.19.1.1 |
| Test | **PASS rc=0** | **PASS rc=0** | **PASS rc=0** |
| xCAT installed | 2.18.1 | 2.18.1 | 2.18.1 |
| `rpm -Kv` on core + dep RPMs | all OK | all OK | all OK |
| `does not verify: no digest` | **none** | **none** | **none** |
